components/brave_wallet/browser/solana_provider_impl.cc

```diff
@@ -25,18 +45,42 @@ void SolanaProviderImpl::Init(
 
 void SolanaProviderImpl::Connect(absl::optional<base::Value> arg,
```

Comment:
As discussed offline, we need more test coverage of the expected API behavior of this function with permission and connected status to avoid regression in the future.

Reply:
Case added in 50bded0d546f7ab08bdc6c82b68166f895984d3b